### DISCUSSION TITLE
I'm considering heavily relying on this project for ArchiveBox.io and want to learn more

### DISCUSSION THREAD

**Discussion by pirate** [boxine/django-huey-monitor#63] [General]
Hi @boxine team!

Thanks for making this awesome Huey monitor open source! It's clearly one of the best options for feature-complete job/worker handling in the Django ecosystem right now, and I'm sure it will grow quickly!

I have a large existing Django project https://github.com/ArchiveBox/ArchiveBox and I'm looking for a new async job system to move our project onto. Huey + your monitor is our highest candidate right now as it supports SQLite, nested tasks with hierarchy and global mutexes, live progress info, and provides a nice UI in the django admin.

If we do this refactor, we'll likely become big contributors to this project (and I'd be happy to send a subset of our donations your way).

I'd love to chat with any of the devs who've worked on this project to get more context on your goals with it, hear about any lessons you've learned with Huey while building this, and see how we can help out if it's a good fit.

If you're down with a quick 20min video call, that would be ideal https://calendly.com/nicksweeting/choose-a-time, otherwise happy to just continue the discussion here too.

**Comment by radusuciu** [comment]
@pirate Have you made a decision to run with this project?

**Comment by pirate** [comment]
Not yet, it's going to be a slow moving refactor as it's a bit change and it's just a spare time project for me, probably on the order of ~6mo.

**Comment by flbraun** [comment]
Hi @pirate!
First of all, sorry for the late response. We just came out of holiday season in Germany, so there was barely anyone around to pick your ticket up.
We would be glad to become part of ArchiveBox, however we must set expectations straight.

django-huey-monitor was born off the need for a Huey-equivalent to [flower](https://github.com/mher/flower). We came up with a prototype of it, tested it in internal use and were happy with its capabilities. We figured we came up with a fairly generic solution, so we put it up on GitHub 'as-is' as a payback to the open source community.
We had some new feature ideas (e.g. XHR-updating admin view, REST API for external systems, auto-vacuuming of old records, etc.), but we a) never had an urgent need for them and b) we got a lot of other things to do, so most of them were scrapped or put on hold.

Long story short: We are happy about any contribution to django-huey-monitor, under the condition that it maintains its 'usuable for everyone' spirit. When a major faction starts contributing you run at risk of tilting towards their specific needs and ideas, but we definitely want django-huey-monitor to keep its identity. If you're unsure about whether we would like a change or not, just open a ticket for discussion before you start implementing. In case you don't feel like our ideas aren't compatible any more, there's still the option to fork (as long as it honors our [license](https://github.com/boxine/django-huey-monitor/blob/master/LICENSE)).

PS: @boxine is a well-funded company, keep your donations. :smile:

**Comment by pirate** [comment]
@Chiron1991 no worries for the delay, I'm on vacation as well and Archivebox is generally slow moving. Thanks for this response, this is perfect, exactly the context I was hoping for!

I will dig deeper when I return from my road trip in October!

**Comment by pirate** [comment]
It's official, archivebox is now using django-huey-monitor as of v0.8!

https://github.com/ArchiveBox/ArchiveBox/tree/dev/archivebox/

Thanks for all your work on this project, it's awesome and I cant wait to keep building on it.
